daemon/images/image_delete.go

```diff
@@ -405,6 +423,10 @@ func (i *ImageService) checkImageDeleteConflict(imgID image.ID, mask conflictTyp
 // imageIsDangling returns whether the given image is "dangling" which means
 // that there are no repository references to the given image and it has no
 // child images.
-func (i *ImageService) imageIsDangling(imgID image.ID) bool {
-	return !(len(i.referenceStore.References(imgID.Digest())) > 0 || len(i.imageStore.Children(imgID)) > 0)
+func (i *ImageService) imageIsDangling(ctx context.Context, imgID image.ID) (bool, error) {
+	children, err := i.imageStore.Children(ctx, imgID)
+	if err != nil {
+		return false, err
```

Comment:
Not with the current implementation, but it's possible future implementations could elect to do so

Reply:
It would only make sense for `NotFound()` to be returned if `imgID` is not found IMO, and in that case there wouldn't be an image to block removal of.